src/mumble/MumbleApplication.cpp

```diff
@@ -56,43 +56,27 @@ bool MumbleApplication::event(QEvent *e) {
 }
 
 #ifdef Q_OS_WIN
-/// gswForward forwards a native Windows keyboard/mouse message
-/// into GlobalShortcutWin's event stream.
-///
-/// @return  Returns true if the forwarded event was suppressed
-///          by GlobalShortcutWin. Otherwise, returns false.
-static bool gswForward(MSG *msg) {
-	GlobalShortcutWin *gsw = static_cast< GlobalShortcutWin * >(GlobalShortcutEngine::engine);
+bool MumbleApplication::nativeEventFilter(const QByteArray &, void *message, long *) {
+	if (QThread::currentThread() != thread()) {
+		return false;
+	}
```

Comment:
I assume if this if was passed, this would mean that someone has made a (serious) programming error. If that is correct, I would change this into a hard assertion that terminates the program once hit.

Reply:
I actually don't know what the original intention was.

I would just remove the code though, as the event loop in `QApplication` is processed by a single thread anyway.

Also, if the thread was different nothing bad would happen.

Reply:
Alright sounds good as well :+1: